src/Uno.UI/UI/Xaml/FrameworkElement.cs

```diff
@@ -268,7 +268,10 @@ protected virtual void OnStyleChanged(Style oldStyle, Style newStyle)
 			{
 				ApplyDefaultStyle();
 
-				if (FeatureConfiguration.FrameworkElement.ClearPreviousOnStyleChange)
+				if (FeatureConfiguration.FrameworkElement.ClearPreviousOnStyleChange && 
+					// Don't clear the default Style, which should always be present.
+					!(bool)(oldStyle == Style.DefaultStyleForType(GetType()))
```

Comment:
@Dissolute Why is this cast as a bool?

Reply:
Good question... I had it as a variable and then pressed the 'inline' tool, and evidently didn't check after